lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -1455,12 +1455,13 @@ private Task completeTask(String taskId, boolean isForced)
       throws TaskNotFoundException, InvalidOwnerException, InvalidStateException,
           NotAuthorizedException {
     String userId = taskanaEngine.getEngine().getCurrentUserContext().getUserid();
-    TaskImpl task = (TaskImpl) this.getTask(taskId);
-    if (reviewRequiredManager.reviewRequired(task)) {
-      return requestReview(taskId);
-    }
+    TaskImpl task;
```

Comment:
Why don't you use `this.getTask(taskId);` here anymore? It looks cleaner to me, because the Task that is returned at the end cannot be undefined this way

Reply:
We've introduced this in #1986. The `getTask(taskId)` operation should be executed within the same transaction stable as `completeTask(..)`. Therefore I moved it back in the try {} catch {} statement.

I realized this during the implementation of this PR